Sprint-3/todo-list/script.js

```diff
@@ -16,10 +54,32 @@ populateTodoList(todos);
 function addNewTodo(event) {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
-  // Write your code here... and remember to reset the input field to be blank after creating a todo!
+  let todoInput = document.getElementById("todo-input");
+  let task = todoInput.value;
+  let todoDeadline = document.getElementById("todo-deadline");
+  let deadline = todoDeadline.value;
+
+
+  if (task.trim() !== "") {
+    todos.push({ task: task, completed: false, deadline: deadline || null });
```

Comment:
Could also consider sanitise the value of `deadline` before adding it to `todos` (in order to always keep the application's states valid).